service/internal/idpplugin/keycloak_plugin.go

```diff
@@ -58,14 +58,47 @@ func EntityResolution(ctx context.Context,
 		slog.DebugContext(ctx, "Lookup", "entity", ident.GetEntityType())
 		var keycloakEntities []*gocloak.User
 		var getUserParams gocloak.GetUsersParams
+		var getClientParams gocloak.GetClientsParams
 		exactMatch := true
 		switch ident.GetEntityType().(type) {
 		case *authorization.Entity_ClientId:
 			slog.DebugContext(ctx, "GetClient", "client_id", ident.GetClientId())
 			clientID := ident.GetClientId()
-			clients, err := connector.client.GetClients(ctx, connector.token.AccessToken, kcConfig.Realm, gocloak.GetClientsParams{
+			getClientParams = gocloak.GetClientsParams{
 				ClientID: &clientID,
-			})
+			}
+		case *authorization.Entity_EmailAddress:
+			getUserParams = gocloak.GetUsersParams{Email: func() *string { t := ident.GetEmailAddress(); return &t }(), Exact: &exactMatch}
+		case *authorization.Entity_UserName:
+			getUserParams = gocloak.GetUsersParams{Username: func() *string { t := ident.GetUserName(); return &t }(), Exact: &exactMatch}
+		case *authorization.Entity_Jwt:
+			entityJwt := ident.GetJwt()
+			// decode the jwt
+			_, mapClaims, err := connector.client.DecodeAccessToken(ctx, entityJwt, kcConfig.Realm)
+			if err != nil {
+				return &authorization.IdpPluginResponse{},
+					status.Error(codes.Internal, db.ErrTextCreationFailed)
+			}
+			// if it has client ID use that
+			clientID, ok := (*mapClaims)["client_id"].(string)
```

Comment:
@mkleene you found an oddity with `client-id`, I believe.  Is this correct logic for Keycloak?

Reply:
according to this: https://datatracker.ietf.org/doc/html/rfc9068, it's required, but there are lots of cases where it's not included

Reply:
and that's a proposed standard so it makes sense